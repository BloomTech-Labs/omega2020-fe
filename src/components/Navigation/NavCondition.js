import React from 'react';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import NavBarMobile from './NarvBar/NavBar-mobile';
import NavBarWebApp from './NarvBar/NavBar-web_app';
import NavBarWeb from './NarvBar/NavBar-web';

const NavCondition = () => {
  const viewPort = false;

  const sm = useMediaQuery('(min-width: 980px)');
  const md = useMediaQuery('(min-width: 1500px)');
  if (viewPort === sm) {
    return <NavBarMobile />;
  } else if (viewPort === md) {
    return <NavBarWebApp />;
  } else {
    return <NavBarWeb />;
  }
};

export default NavCondition;
