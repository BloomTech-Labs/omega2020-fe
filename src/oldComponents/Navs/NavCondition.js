import React, {useState, useEffect} from 'react';

import LoggedIn from './LoggedIn';
import LoggedOut from './LoggedOut';

const NavCondition = (props) => {
    if (props.token === null) {
        return <LoggedOut />
    } else {
        return <LoggedIn />
    }
}

export default NavCondition;