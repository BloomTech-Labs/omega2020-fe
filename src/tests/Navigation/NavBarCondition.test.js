import React from 'react';
import renderer from 'react-test-renderer';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import '@testing-library/jest-dom/extend-expect';
import { Box, Container } from '@material-ui/core';
import NavCondition from '../../components/Navigation/NavBarCondition';

configure({ adapter: new Adapter() });


describe('the length of the material ui and match the snapshot of the foot mobile component', () => {

    const wrapper = mount(<NavCondition />)

    it('snapshot the component', () => {
        const menuShot = renderer.create(<NavCondition />).toJSON();
        expect(menuShot).toMatchSnapshot();
    })

    it('render the length of the material ui', () => {
        expect(wrapper.find(Box).length).toEqual(7);
        expect(wrapper.find(Container).length).toEqual(1);
    })

})