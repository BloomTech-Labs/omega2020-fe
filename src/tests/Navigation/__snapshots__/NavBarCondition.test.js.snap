// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the length of the material ui and match the snapshot of the foot mobile component snapshot the component 1`] = `
<div
  className="MuiBox-root MuiBox-root-19 makeStyles-root-1"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      className="MuiBox-root MuiBox-root-20 makeStyles-root-3"
    >
      <div
        className="MuiBox-root MuiBox-root-21 makeStyles-leftContainer-4"
      >
        <div
          className="MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-content"
          />
          <div
            className="MuiCardHeader-action"
          >
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  style={
                    Object {
                      "fontSize": 30,
                    }
                  }
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 18c3.31 0 6-2.69 6-6s-2.69-6-6-6-6 2.69-6 6 2.69 6 6 6z"
                    opacity=".3"
                  />
                  <path
                    d="M12 20c4.42 0 8-3.58 8-8s-3.58-8-8-8-8 3.58-8 8 3.58 8 8 8zm0-14c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6 2.69-6 6-6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-22 makeStyles-centerContainer-6"
      >
        <div
          className="MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-avatar"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-root-3 MuiTypography-colorPrimary"
              href="/menu"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <img
                alt="logo"
                src="omega-logo.png"
                style={
                  Object {
                    "height": 30,
                    "width": 30,
                  }
                }
              />
            </a>
          </div>
          <div
            className="MuiCardHeader-content"
          />
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-23 makeStyles-rightContainer-5"
      >
        <div
          className="MuiBox-root MuiBox-root-24 makeStyles-icons-7"
        >
          <div
            className="MuiCardHeader-root"
          >
            <div
              className="MuiCardHeader-avatar"
            >
              <div
                className="MuiBox-root MuiBox-root-25"
              >
                <div
                  aria-haspopup="true"
                  className="MuiAvatar-root MuiAvatar-circle makeStyles-color-17 MuiAvatar-colorDefault"
                  onClick={[Function]}
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="MuiCardHeader-content"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <hr
    style={
      Object {
        "opacity": 0.2,
      }
    }
  />
</div>
`;
